include/graaflib/algorithm/shortest_path.tpp

```diff
@@ -111,4 +111,47 @@ std::optional<graph_path<WEIGHT_T>> dijkstra_shortest_path(
   return reconstruct_path(start_vertex, end_vertex, vertex_info);
 }
 
+template <typename V, typename E, graph_type T, typename WEIGHT_T>
+[[nodiscard]] std::unordered_map<vertex_id_t, graph_path<WEIGHT_T>>
+dijkstra_shortest_paths(const graph<V, E, T>& graph,
+                        vertex_id_t source_vertex) {
+  std::unordered_map<vertex_id_t, graph_path<WEIGHT_T>> shortest_paths;
+
+  std::priority_queue<std::pair<WEIGHT_T, vertex_id_t>,
+                      std::vector<std::pair<WEIGHT_T, vertex_id_t>>,
+                      std::greater<>>
+      to_explore;
```

Comment:
This was a 50:50 thing for me, especially since the `prev_id` is unused then. I added it for now, but I think this is something to revisit in the future.